ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdStub.java

```diff
@@ -0,0 +1,376 @@
+
+  void setVungleListener(@Nullable VungleListener vungleListener) {
+    this.mVungleListener = new WeakReference<>(vungleListener);
+  }
+
+  @Nullable
+  private VungleListener getVungleListener() {
+    return mVungleListener.get();
+  }
+
+  boolean isActive() {
+    return mAdLayout.get() != null;
+  }
+
+  void requestBannerAd(@NonNull Context context, @NonNull String appId) {
+    Log.d(TAG, "requestBannerAd: " + this);
+    mPendingRequestBanner = true;
+    VungleInitializer.getInstance()
+        .initialize(
+            appId,
+            context.getApplicationContext(),
+            new VungleInitializer.VungleInitializationListener() {
+              @Override
+              public void onInitializeSuccess() {
+                loadBanner();
+              }
+
+              @Override
+              public void onInitializeError(String errorMessage) {
+                Log.d(TAG, "SDK init failed: " + VungleBannerAdStub.this);
+                VungleListener listener = getVungleListener();
+                VungleManager.getInstance().removeActiveBannerAd(mPlacementId);
+                if (mPendingRequestBanner && listener != null) {
+                  listener.onAdFailedToLoad(AdRequest.ERROR_CODE_INTERNAL_ERROR);
+                }
+              }
+            });
+  }
+
+  void destroy(@Nullable View adView) {
+    Log.d(TAG, "Vungle banner adapter try to destroy:" + this);
+    if (adView == mAdLayout.get()) {
+      Log.d(TAG, "Vungle banner adapter destroy:" + this);
+      mVisibility = false;
+      VungleManager.getInstance().removeActiveBannerAd(mPlacementId);
+      cleanUp();
+      mPendingRequestBanner = false;
+    }
+  }
+
+  /**
+   * This method is a workaround for banner leak issue, and most callers should use {@link
+   * VungleBannerAdStub#destroy(View)}.
+   */
+  void destroy() {
+    destroy(mAdLayout.get());
+  }
+
+  void cleanUp() {
+    Log.d(TAG, "Vungle banner adapter try to cleanUp:" + this);
+
+    if (mVungleBannerAd != null) {
+      Log.d(TAG, "Vungle banner adapter cleanUp: destroyAd # " + mVungleBannerAd.hashCode());
+      mVungleBannerAd.destroyAd();
+      detach();
+      mVungleBannerAd = null;
+    }
+
+    if (mVungleNativeAd != null) {
+      Log.d(
+          TAG, "Vungle banner adapter cleanUp: finishDisplayingAd # " + mVungleNativeAd.hashCode());
+      mVungleNativeAd.finishDisplayingAd();
+      detach();
+      mVungleNativeAd = null;
+    }
+  }
+
+  void preCache() {
+    if (AdConfig.AdSize.isBannerAdSize(mAdConfig.getAdSize())) {
+      Banners.loadBanner(mPlacementId, mAdConfig.getAdSize(), null);
+    } else {
+      Vungle.loadAd(mPlacementId, null);
+    }
+  }
+
+  void updateVisibility(boolean visible) {
+    this.mVisibility = visible;
+    if (mVungleBannerAd != null) {
+      mVungleBannerAd.setAdVisibility(visible);
+    }
+    if (mVungleNativeAd != null) {
+      mVungleNativeAd.setAdVisibility(visible);
+    }
+  }
+
+  private LoadAdCallback mAdLoadCallback =
```

Comment:
could we move mAdLoadCallback and mAdPlayCallback, and have fuction  to "stub" to call 
"VungleManager.getInstance().removeActiveBannerAd(mPlacementId);" ? Or even call destroy()

Reply:
sure, have deleted this stub file.